### PR TITLE
add Not to delete!

### PR DESCRIPTION
**Pull request by bkamins**
This is not urgent, but simple and will improve consistency.

### CODE REVIEW

File: src/dataframe/dataframe.jl [hunk 1]
```diff
@@ -791,19 +791,21 @@ function Base.delete!(df::DataFrame, inds)
...
     foreach(col -> deleteat!(col, inds), _columns(df))
-    df
+    return df
 end
 
 function Base.delete!(df::DataFrame, inds::AbstractVector{Bool})
     if length(inds) != size(df, 1)
         throw(BoundsError(df, (inds, :)))
     end
-    drop = findall(inds)
```
Comment (nalimilan) [inline src/dataframe/dataframe.jl:802]: I think this was intentional: computing integer indices once is faster than using logical indexing for each column, as it requires computing the sum again and again. Though a comment wouldn't hurt.

Comment (bkamins) [reply src/dataframe/dataframe.jl:802]: indeed - now I remember thinking about it some tome ago. I will revert. Anyway this PR should not be merged before #2199 as it is conflicting. Thank you!

**Comment by bkamins** [comment]
Thank you. I will rebase it and merge after #2199 is done

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**